src/common/useOutsideEvent.ts

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+
+// Utilizes the useEffect hook to handle when an event is fired on the document, but outside of the specified element
+export default function useOutsideEvent(type: string, ref: React.RefObject<Element>, callback: () => void): void {
+    React.useEffect(() => {
+        function handleEvent(event: Event): void {
+            const containsEventTarget = !!ref?.current?.contains(event.target as Node);
```

Comment:
Looks like this doesn't work in ie11, we may need to add a polyfill -- maybe something like this? https://stackoverflow.com/questions/57606743/ie11-object-doesnt-support-property-or-method-contains

Reply:
Done.